src/hooks/isMobileDevice.tsx

```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+export default function useIsMobileDevice() {
+  const [screenWidth, setscreenWidth] = React.useState<number>(
+    window.innerWidth
+  );
+  React.useEffect(() => {
+    const updateDimension = () => {
+      setscreenWidth(window.innerWidth);
+    };
+    window.addEventListener("resize", updateDimension);
+  }, [screenWidth]);
+  return screenWidth <= 500;
+}
```

Comment:
I added this hook to dynamically check if the device is mobile. 
It eliminates having to do the check in every component I need it in.
Also, using this it now updates the page real-time in responsive mode.